api/client/build.go

```diff
@@ -354,9 +367,16 @@ func getDockerfileRelPath(givenContextDir, givenDockerfile string) (absContextDi
 	}
 
 	// Evaluate symlinks in the path to the Dockerfile too.
-	absDockerfile, err = filepath.EvalSymlinks(absDockerfile)
-	if err != nil {
-		return "", "", fmt.Errorf("unable to evaluate symlinks in Dockerfile path: %v", err)
+	//
+	// FIXME. We use isUNC (always false on non-Windows platforms) to workaround
+	// an issue in golang. On Windows, EvalSymLinks does not work on UNC file
+	// paths (those starting with \\). This hack means that when using links
+	// on UNC paths, they will not be followed.
+	if !isUNC(absContextDir) {
```

Comment:
Oi, good catch.  Since the two are generated from each other, it's likely this just happened to work, but the right thing here should be absDockerfile.  I'll push an update.